doc: Add new stories with tips and tricks (#560)
Doist/reactist: src/new-components/box/box.stories.mdx

import { Story, Meta } from '@storybook/addon-docs/blocks'

<Meta title="Tips and tricks/Using the Box component" />

# Tips and tricks about the `Box` component

## You don't need the `classNames` helper as often now

The new `Box` component's `className` prop receives the same kind of values supported by this helper. So you can do things like this:

```jsx
// This…
<Box className={classNames(styles.container, 'custom-class-name')} />

// …can be written like this

> <Box className={[styles.container, 'custom-class-name']} />

Reviewer: 😍 This is so great!

// Other examples:
<Box
  className={[
    styles.container,
    isReadOnly ? styles.readOnly : null,
    { [styles.disabled]: isDisabled },
  ]}
/>
```

Granted, this is only supported in the `Box`. This is because the `Box` is the only design system component that officially supports custom styles. Remember that for the other components you can use `exceptionallySetClassName`, but this one is an escape hatch, that we should aim to not use as much as the design system matures. So this one does not support the feature.

## The `paddingX` and `paddingY` shorthand props

> The `Box` component has props to set its padding in all four direction: `paddingTop`, `paddingRight`, `paddingBottom`, `paddingLeft`. You probably have noticed that we can set them all at once with `padding`.

Reviewer: ```suggestion
The `Box` component has props to set its padding in all four direction: `paddingTop`, `paddingRight`, `paddingBottom`, `paddingLeft`. You have probably noticed that we can set them all at once with `padding`.
```

Author: Ricardo suggested another unrelated edit to this very line, so I'll apply your suggestion alongside Ricardo's suggestion.

> But you probably missed the fact that we can set the padding in one of the two dimensions at once as well. This is achieved with `paddingX` (which sets `paddingLeft` and `paddingRight`) and `paddingY` (which sets `paddingTop` and `paddingBottom`).

Reviewer: Super minor note just to remove repeating `probably`
```suggestion
But you might have missed the fact that we can set the padding in one of the two dimensions at once as well. This is achieved with `paddingX` (which sets `paddingLeft` and `paddingRight`) and `paddingY` (which sets `paddingTop` and `paddingBottom`).
```

> The `Box` component has props to set its padding in all four direction: `paddingTop`, `paddingRight`, `paddingBottom`, `paddingLeft`. You probably have noticed that we can set them all at once with `padding`.
> 
> But you probably missed the fact that we can set the padding in one of the two dimensions at once as well. This is achieved with `paddingX` (which sets `paddingLeft` and `paddingRight`) and `paddingY` (which sets `paddingTop` and `paddingBottom`).

Author: I'm combining here your suggestion with another change suggested by Paul below:

```suggestion
The `Box` component has props to set its padding in all four direction: `paddingTop`, `paddingRight`, `paddingBottom`, `paddingLeft`. You have probably noticed that we can set them all at once with `padding`. But you probably missed the fact that we can set the padding in one of the two dimensions at once as well. This is achieved with `paddingX` (which sets `paddingLeft` and `paddingRight`) and `paddingY` (which sets `paddingTop` and `paddingBottom`).
```
